app/models/compute_resource.rb

```diff
@@ -457,6 +457,19 @@ def associate_by(name, attributes)
       first
   end
 
+  def associated_vm(host)
+    vms(:eager_loading => true).find { |vm| associate_by_host("mac", vm.interfaces.map(&:mac), host) }
```

Comment:
Not sure how this will behave - do we need/want eager loading here? I understand that we might not want to have a query for every single vm :thinking: